pkg/config/types/v1alpha1/v1alpha1_provider.go

```diff
@@ -406,17 +418,102 @@ func (e *EtcdConfig) ExtraArgs() map[string]string {
 
 // Mirrors implements the Registries interface.
 func (r *RegistriesConfig) Mirrors() map[string]config.RegistryMirrorConfig {
-	return r.RegistryMirrors
+	mirrors := make(map[string]config.RegistryMirrorConfig, len(r.RegistryMirrors))
+
+	for k, v := range r.RegistryMirrors {
+		mirrors[k] = v
+	}
+
+	return mirrors
 }
 
 // Config implements the Registries interface.
 func (r *RegistriesConfig) Config() map[string]config.RegistryConfig {
-	return r.RegistryConfig
+	registries := make(map[string]config.RegistryConfig, len(r.RegistryConfig))
+
+	for k, v := range r.RegistryConfig {
+		registries[k] = v
+	}
+
+	return registries
 }
 
-// ExtraFiles implements the Registries interface.
-func (r *RegistriesConfig) ExtraFiles() ([]config.File, error) {
-	return criplugin.GenerateRegistriesConfig(r)
+// TLS implements the Registries interface.
+func (r *RegistryConfig) TLS() config.RegistryTLSConfig {
+	if r == nil {
+		return nil
+	}
+
+	return r.RegistryTLS
+}
+
+// Auth implements the Registries interface.
+func (r *RegistryConfig) Auth() config.RegistryAuthConfig {
+	if r == nil {
+		return nil
+	}
+
+	return r.RegistryAuth
+}
+
+// Username implements the Registries interface.
+func (r *RegistryAuthConfig) Username() string {
+	return r.RegistryUsername
+}
+
+// Password implements the Registries interface.
+func (r *RegistryAuthConfig) Password() string {
+	return r.RegistryPassword
+}
+
+// Auth implements the Registries interface.
+func (r *RegistryAuthConfig) Auth() string {
+	return r.RegistryAuth
+}
+
+// IdentityToken implements the Registries interface.
+func (r *RegistryAuthConfig) IdentityToken() string {
+	return r.RegistryIdentityToken
+}
+
+// ClientIdentity implements the Registries interface.
+func (r *RegistryTLSConfig) ClientIdentity() *x509.PEMEncodedCertificateAndKey {
+	return r.TLSClientIdentity
+}
+
+// CA implements the Registries interface.
+func (r *RegistryTLSConfig) CA() []byte {
+	return r.TLSCA
+}
+
+// InsecureSkipVerify implements the Registries interface.
+func (r *RegistryTLSConfig) InsecureSkipVerify() bool {
+	return r.TLSInsecureSkipVerify
+}
+
+// GetTLSConfig prepares TLS configuration for connection.
```

Comment:
not for this PR, but some TODO later - this can be extracted out of `v1apha1` to be some common function operating on the interface and returning `*.tls.Config` from it.

Reply:
:+1: